src/main/pileup.js

```diff
@@ -80,6 +80,37 @@ function create(elOrId: string|Element, params: PileupParams): Pileup {
       ReactDOM.render(<Root referenceSource={referenceTrack.source}
                             tracks={vizTracks}
                             initialRange={params.range} />, el);
+
+  //if the element doesn't belong to document DOM observe DOM to detect
```

Comment:
You can do that, but for me this code is really hard to read and understand. This later on will influence time spent on debuging and improving code.

But this is my opinion. Maybe for Javascript developers this syntax is obvious and easy to read :).